### PR TITLE
feat: Add Language#normalizeLanguageOptions()

### PR DESCRIPTION
**Pull request by nzakas**
#### Prerequisites checklist

-   [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/HEAD/CONTRIBUTING.md).

#### What is the purpose of this pull request?

Add `normalizeLanguageOptions` to the `Language` type.

#### What changes did you make? (Give an overview)

Added an optional `normalizeLanguaeOptions()` method to the `Language` type.

#### Related Issues

refs https://github.com/eslint/eslint/issues/19037

### CODE REVIEW

**Review by mdjermanovic — APPROVED** [review]
LGTM, thanks! Not sure if we should wait for the implementation of this feature before merging this.

**Comment by nzakas** [comment]
Probably a good idea to avoid confusion. I wish GitHub had a "schedule" feature.

**Review by fasttime — APPROVED** [review]
LGTM, thanks. Leaving open until eslint/eslint#19104 is ready to merge.

**Pull request merged**
